source/main.cc

```diff
@@ -412,92 +434,68 @@ int main (int argc, char *argv[])
     {
       deallog.depth_console(0);
 
-      // print the basic header from processor 0
-      if (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 0)
-        print_aspect_header(std::cout);
+      int current_idx = 1;
+      bool i_am_proc_0 = (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 0);
 
+      std::string prm_name = "";
+      bool print_xml = false;
 
-      // check whether ASPECT was called with exactly one argument. if not,
-      // print an error message
-      //
-      // however, this does not work with PETSc because for PETSc, one
-      // may pass any number of flags on the command line; unfortunately,
-      // the PETSc initialization code (run through the call to
-      // MPI_InitFinalize above) does not filter these out, so all we
-      // can test is that there is *at least* one argument on the command
-      // line
-      if (
-        (
-#ifdef ASPECT_USE_PETSC
-          argc < 2
-#else
-          argc != 2
-#endif
-        )
-        // or if the user called aspect with either --help or -h as
-        // the sole arguments
-        ||
-        (
-          (
-#ifdef ASPECT_USE_PETSC
-            argc >= 2
-#else
-            argc == 2
-#endif
-          )
-          &&
-          (
-            (std::string(argv[1]) == "--help")
-            ||
-            (std::string(argv[1]) == "-h")
-          )
-        )
-      )
+      while (current_idx<argc)
         {
-          // print usage info only on processor 0
-          if (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 0)
+          std::string arg = std::string(argv[current_idx++]);
```

Comment:
no need for the explicit conversion on the rhs, just `argv[current_idx++]` should work.

I'm not a fan of hiding the counter increment in this rather obscure place, though